### PR TITLE
Support the log warning if the user has write access to the data

### PR DESCRIPTION
**Pull request by Jayrmajithia**

### CODE REVIEW

**Comment by Jayrmajithia** [comment]
it has been refactored now it has check_user as a trait implementation on the backend. For other databases, the idea of retrieving the SQL need to be changed.

File: tesseract-clickhouse/src/lib.rs [hunk 1]
```diff
@@ -84,5 +84,31 @@ impl Backend for Clickhouse {
...
+            .get_handle()
+            .and_then(move |c| c.query(index_sql).fetch_all())
+            .and_then(move |(_, block): (_, Block<Complex>)| {
+                let values: Vec<&str> = block.get(0, "values")?;
+                let names: Vec<&str> = block.get(0, "names")?;
+                if names.contains(&"readonly") && names.contains(&"allow_ddl") {
+                    let rd_index = names.iter().position(|&r| r=="readonly").unwrap();
+                    let ad_index = names.iter().position(|&r| r=="allow_ddl").unwrap();
+                    // let rd_value = values.get(rd_index).parse::<u32>().unwrap();
+                    // let ad_value = values.get(ad_index).parse::<u32>().unwrap();
```
Comment (MarcioPorto) [inline tesseract-clickhouse/src/lib.rs]: Remove these if they are not being used

**Comment by MarcioPorto** [comment]
@jspeis Is this ready to merge?

File: tesseract-core/src/backend.rs [hunk 2]
```diff
@@ -38,6 +38,10 @@ pub trait Backend {
             &query_ir.growth,
         )
     }
+
+    fn check_user(&self) -> Box<dyn Future<Item=(), Error= Error>> {
+        unimplemented!()
```
Comment (jspeis) [inline tesseract-core/src/backend.rs]: One thing we might want to do by default here would be to log that this is unimplemented but maybe not error? thoughts?

Comment (MarcioPorto) [reply tesseract-core/src/backend.rs]: Yeah, that's definitely better than erroring out.

**Comment by MarcioPorto** [comment]
Closing in favor of https://github.com/tesseract-olap/tesseract/pull/287

**Pull request closed without merging**